aten/src/ATen/native/cuda/ScanUtils.cuh

```diff
@@ -57,14 +57,18 @@ __device__ void binary_op_update(const scalar_t lhs, scalar_t& rhs, const idx_t
  * Each thread block processes one or more sets of contiguous rows (processing multiple rows
  * per thread block is quicker than processing a single row, especially for short rows).
  */
-template<typename scalar_t, int num_threads_x, int num_threads_y, class BinaryFunction>
+template<typename scalar_t, class BinaryFunction>
 __global__ void tensor_kernel_scan_innermost_dim_with_indices(const scalar_t *self_, scalar_t *values_, int64_t *indices_,
                                                 int num_rows, int row_size,
+                                                const uint32_t num_threads, const uint32_t log_num_threads_x,
                                                 scalar_t init, BinaryFunction binary_op) {
-  __shared__ scalar_t vbuf[num_threads_y][2 * num_threads_x];
-  __shared__ int64_t ibuf[num_threads_y][2 * num_threads_x];
-  scalar_t* row_buf = vbuf[threadIdx.y];
-  int64_t* row_idx_buf = ibuf[threadIdx.y];
+  // dynamic memory allocation for vbuf and ibuf
+  extern __shared__ char buf[];
```

Comment:
with some compiler versions you need  `alignas(sizeof(double)) extern __shared__ char buf[];`, otherwise compiler will complain of insufficient alignment

Reply:
Done, thanks!